lib/pf/activation.pm

```diff
@@ -501,7 +514,10 @@ sub create_and_send_activation_code {
     my ($mac, $pid, $pending_addr, $template, $type, $portal, %info) = @_;
 
     my ($success, $err) = ($TRUE, 0);
-    my $activation_code = create($mac, $pid, $pending_addr, $type, $portal);
+    my %args = ( 
+	mac 	=> $mac,  pid 	 => $pid, 	pending => $pending_addr, 
```

Comment:
I disagree in this case.
It makes the subroutine too long.
A tabular format is better if there are a lot of arguments.
Conciseness is also a virtue.